### PR TITLE
Update Doctrine ORM and DBAL

### PR DESCRIPTION
**Pull request by Sperling-0**
- Fix address type display bug
- Adapt classes to updated Doctrine API 
 
Ticket: https://phabricator.wikimedia.org/T356744

### CODE REVIEW

**Review by gbirke — COMMENTED** [review]
You're on the right track overall, the next step would be to change the type of the `AddressChange::addressType` property from string to `AddressType` and fix all the places where a string constant is used instead of `AddressType`.

Another step that needs to be done is wrapping the `AddressTypeContextFactory` inside the `TestAddressTypeContextFactory` and call `registerCustomTypes` method in the setup. This will make the new type available in the tests (otherwise all database-related tests will fail because Doctrine doesn't know about the type). You can look at how the other repositories (e.g. donations or payments) wrap the factory in the test factory and call `registerCustomTypes`

File: config/DoctrineClassMapping/WMDE.Fundraising.AddressChangeContext.Domain.Model.Address.dcm.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <id name="id" type="integer" column="id">
       <generator strategy="IDENTITY"/>
     </id>
+    <field name="addressType" type="AddressType" column="address_type" length="8"/>
```
[the remark this answers is not in the document]
Comment (Sperling-0) [reply config/DoctrineClassMapping/WMDE.Fundraising.AddressChangeContext.Domain.Model.Address.dcm.xml]: Done.

File: src/Domain/Model/AddressType.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+declare( strict_types=1 );
+
+namespace WMDE\Fundraising\AddressChangeContext\Domain\Model;
+
+enum AddressType: string {
```
Comment (Abban) [inline src/Domain/Model/AddressType.php:6]: @gbirke requested that the Donation address type enum not be string backed: https://github.com/wmde/fundraising-donations/pull/233

So I think we should do the same here to keep is consistent.

Comment (moiikana) [reply src/Domain/Model/AddressType.php:6]: @Sperling-0 as you were having questions about this earlier, the enum should should ideally look like the DonorType here, so you can remove the strings https://github.com/wmde/fundraising-donations/blob/3f103d1355d238da76a185d8d6572b45f795f001/src/Domain/Model/DonorType.php

**Comment by moiikana** [comment]
once this is merged, create a new release

**Review by moiikana — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -4,14 +4,15 @@
 	"license": "GPL-2.0-or-later",
 	"require": {
 		"php": ">=8.1",
-		"doctrine/orm": "^2.11",
-		"doctrine/dbal": "^3.3",
+		"doctrine/orm": "~2.18 | ~3.0",
```
Comment (Abban) [inline composer.json:7]: How come we accept different versions here?

If it's to please Travis we should be able to use the more recent ones

**Review by Abban — APPROVED** [review]

File: composer.json (hunk 3 above)
Comment (Sperling-0) [reply composer.json:7]: We can ask Gabriel about this when he is back.

**Pull request merged**